### PR TITLE
add gaurd to stop injest call if data is falsey

### PR DESCRIPTION
**Pull request by JDragovichAlertLogic**
### Problem Description
Sending a blank post body to ingest was causing ingest to return a 400.

### Solution Description
in the `send` function, check if the data is falsey. If it is, just return the callback.

### CODE REVIEW

File: al_aws_collector.js [hunk 1]
```diff
@@ -203,7 +203,14 @@ class AlAwsCollector {
     send(data, callback){
         var collector = this;
         var ingestType = collector._ingestType;
-        
+
+        // if the data is falsey, just return the callback
+        if(!data){
+            console.warn('data appears to be empty, skipping send to ingest');
```
Comment (kkuzmin) [inline al_aws_collector.js]: I think we don't need this warn message

File: al_aws_collector.js [hunk 2]
```diff
@@ -203,7 +203,14 @@ class AlAwsCollector {
...
+            callback(null);
+            return;
```
Comment (kkuzmin) [inline al_aws_collector.js]: Should we use `return callback(null);` instead of two statements here?

File: al_aws_collector.js [hunk 3]
```diff
@@ -203,7 +203,14 @@ class AlAwsCollector {
     send(data, callback){
         var collector = this;
         var ingestType = collector._ingestType;
-        
+
+        // if the data is falsey, just return the callback
+        if(!data){
```
Comment (kkuzmin) [inline al_aws_collector.js]: Let's also check for an empty list/object as well

**Comment by JDragovichAlertLogic** [comment]
Closing this because it was not in my fork. My bad

**Pull request closed without merging**